src/server.c

```diff
@@ -3531,6 +3586,8 @@ void call(client *c, int flags) {
         char *latency_event = (real_cmd->flags & CMD_FAST) ?
                                "fast-command" : "command";
         latencyAddSampleIfNeeded(latency_event,duration/1000);
+        if (!(c->flags & CLIENT_MULTI))
+            durationAddSample(EL_DURATION_TYPE_CMD, duration);
```

Comment:
I was expecting we would check for recursive commands here, doesn't EVAL and modules that issue RM_CALL() also double count now?

Reply:
Yes, and FCALL.

Reply:
again, i'm ashamed i have missed it.
so checking `server.execution_nesting==1` should fix both EXEC and EVAL, we well as RM_Call from within a module command.
It would also avoid measuring RM_Call from a thread or a timer event (since it'll be incremented when a module context is created and again in `call()`), which i suppose is good (i.e. module actions are in anyway "internal" and not "commands" from the perspective of the event loop.